internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -474,47 +474,52 @@ func (provisioner *KopsProvisioner) ProvisionCluster(cluster *model.Cluster, aws
 		}
 	}
 
+	wait = 240
 	supportAppsWithDaemonSets := map[string]string{
 		"calico-node":                  "kube-system",
 		"k8s-spot-termination-handler": "kube-system",
 	}
 	for daemonSet, namespace := range supportAppsWithDaemonSets {
 		if daemonSet == "k8s-spot-termination-handler" && (len(os.Getenv(model.MattermostChannel)) > 0 || len(os.Getenv(model.MattermostWebhook)) > 0) {
+			logger.Infof("Waiting up to %d seconds for %q daemonset to get it...", wait, daemonSet)
+			ctx, cancel := context.WithTimeout(context.Background(), time.Duration(wait)*time.Second)
+			defer cancel()
 			daemonSetObj, err := k8sClient.Clientset.AppsV1().DaemonSets(namespace).Get(ctx, daemonSet, metav1.GetOptions{})
 			if err != nil {
-				return errors.Wrapf(err, "failed to get daemonSet %s", daemonSet)
+				return errors.Wrapf(err, " failed to get daemonSet %s", daemonSet)
 			}
-
 			var payload []k8s.PatchStringValue
-			for i, envVar := range daemonSetObj.Spec.Template.Spec.Containers[0].Env {
-				if envVar.Name == "CLUSTER" {
-					payload = []k8s.PatchStringValue{{
-						Op:    "replace",
-						Path:  "/spec/template/spec/containers/0/env/" + strconv.Itoa(i) + "/value",
-						Value: cluster.ID,
-					}}
-				}
-				if envVar.Name == "MATTERMOST_CHANNEL" && len(os.Getenv(model.MattermostChannel)) > 0 {
-					payload = append(payload,
-						k8s.PatchStringValue{
-							Op:    "replace",
-							Path:  "/spec/template/spec/containers/0/env/" + strconv.Itoa(i) + "/value",
-							Value: os.Getenv(model.MattermostChannel),
-						})
-				}
-				if envVar.Name == "MATTERMOST_WEBHOOK" && len(os.Getenv(model.MattermostWebhook)) > 0 {
-					payload = append(payload,
-						k8s.PatchStringValue{
+			if daemonSetObj.Spec.Selector != nil {
```

Comment:
Could you help me understand why we are adding this check?

Reply:
When in a cluster we don't have spot instances at all this selector expected to be nil. So in this case it is faster to bypass this block and continue to the next steps of provisioning.